### PR TITLE
Enhancements for the JPA-related content

### PR DESCRIPTION
**Pull request by MichalMaler**
**Enhancements for the JPA-related content:**

* Removing concept-type information out of the JPA tutorial and creating a new JPA concept module.
* Adding xrefs to related documentation.
* Minor style tweaks in the Quarkus security documentation.

**Main files for downstreaming:**

security-jpa-concept.adoc
security-basic-authentication-tutorial.adoc
security-identity-providers-concept.adoc
security-authentication-mechanisms-concept.adoc

Closing: https://issues.redhat.com/browse/QDOCS-99

### CODE REVIEW

**Comment by sberyozkin** [comment]
@MichalMaler I think it is an excellent write-up, thanks, nicely refactored.
My only ask is to copy (not move) to this new JPA concept doc this section as well, https://quarkus.io/guides/security-basic-authentication-tutorial#define-the-user-entity 

What you have moved to the JPA concept is an extra content in addition to what is covered in https://quarkus.io/guides/security-basic-authentication-tutorial#define-the-user-entity, so copying it will make the JPA concept doc complete

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:120]: Please add Reeferences section here, every sec doc should link back to the main security overview concept and other related concept for the ease of navigation

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:27]: Please copy the section from the Basic Auth and JPA tutorial over here

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:27]: It will cause some duplication but but the Basic Auth and JPA tutorial is about getting started fast with Quarkus sec with the help of Basic Auth and JPA, while the purpose of this doc is to offer a complete JPA concept

Comment (MichalMaler) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:27]: OK @sberyozkin . Adding now. Will rebase and squash.

Comment (MichalMaler) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:27]: Done and done. See below from row 28.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:174]: Added ;) Good idea

**Comment by MichalMaler** [comment]
@sberyozkin Hello Siarhei! Added all the items from your wish list.
@michelle-purcell Michelle, can I have your opinion on newly added content? Much thanks for your expert review :)

**Comment by MichalMaler** [comment]
@jmartisk Thanks Jan for the suggestions and additional SME review during the drafting phase.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:6]: Broken xref. See preview: 

image

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:4]: Is this a good topic to help a developer get started with Quarkus? If so: 

```suggestion
:categories: security,jpa,getting-started
```

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:6]: Fix xref.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:4]: @sberyozkin WDYT?

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:6]: I think the _**s**_ in security should be uppercase here and elsewhere when referring to the noun "Quarkus Security", which is the name of the Quarkus framework for security.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:4]: Hi @michelle-purcell @MichalMaler, on its own is a fairly technical concept, it is not really something that new users should be forwarded to when starting with Quarkus. It does make sense to show JPA in the getting started with quarkus security (Basic plus JPA) just because saving the users info in DB is better than directly in application.properties, but IMHO on its own it is not a getting started material :-). Perhaps at a later stage when a reactive JPA gets supported we can consider how to make it move visible, thanks

**Review by sberyozkin — APPROVED** [review]
LGTM, lets wait for a bit longer before merging in case more review comments will follow

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:77]: Clarify what the 'it' is:
```suggestion
By default, `security-jpa` uses bcrypt-hashed passwords, and you can also configure plain text or custom passwords instead.
```

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:79]: Suggestion:Change _**us**_ to **_you_** to be consistent with previous statements.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:116]: Suggestion to simplify if not breaking the technical meaning:

```suggestion
When developing applications with Quarkus, you can decide how to manage password storage and hashing. You can choose to keep the default password and hashing settings of Quarkus, or you can hash passwords manually.
```

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:4]: @sberyozkin - Thanks for explaining. Sounds good 👍

File: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:4]: Is there a category set up in the docs architecture for `jpa`?

We don't mention one in the doc contributor guide and I don't see it in the drop-down list on the doc portal yet.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:4]: Same comment here about `jpa` too.

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:130]:
```suggestion
The following snippet shows how to set a custom password provider that uses the SHA256 hashing algorithm.
```

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:176]: Links LGTM on preview 💯

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:173]: Remove unrequired article _a_.

```suggestion
However, it is possible to store passwords as plain text with the `@Password(PasswordType.CLEAR)` annotation when operating in a test environment.
```

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (MichalMaler) [reply docs/src/main/asciidoc/security-jpa-concept.adoc:6]: Great catch @michelle-purcell . Fixing here and on some other places too.

**Comment by MichalMaler** [comment]
@michelle-purcell Thank you for the review! Applied lot of great feedback. 
@sberyozkin , please, can you go over last two Michelle's suggestion and provide a correct version of here fix? I like the style correction, however, I am not sure if it didn't break the technical meaning.

Please, put your OK tho the last two comments or provide an accurate formulation. Will apply it and do the last squash/rebase combo.

Cheers.

**Comment by sberyozkin** [comment]
@MichalMaler Hi Michal, the last 2 comments from Michelle look fine to me

**Comment by MichalMaler** [comment]
@sberyozkin Cool! Applied.
My job is done here.
Cheers!

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:171]: Remove article **_a_**. Suggestion: 

```suggestion
For applications running in a production environment, do not store passwords as plain text.
```

File: docs/src/main/asciidoc/security-identity-providers-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-identity-providers-concept.adoc:30]: Add the main Quarkus Security overview topic here:

```suggestion
* xref:security-ldap.adoc[Using security with an LDAP realm]
* xref:security-overview-concept.adoc[Quarkus Security overview]
```

**Review by michelle-purcell — APPROVED** [review]
@MichalMaler - Super job.  Thank you. Looks great 💯 See my style comments ⬆️ .

[I completed my review. Apologies, my laptop is dead so I had to find another machine to finish the review!]

**Comment by sberyozkin** [comment]
Thanks @MichalMaler and @michelle-purcell

File: docs/src/main/asciidoc/security-jpa-concept.adoc
Comment (michelle-purcell) [inline docs/src/main/asciidoc/security-jpa-concept.adoc:20]: @MichalMaler - Sorry just noticed this now while verifying single-sourced content on downstream docs. Should this be an xref? Thinking if not an xref, then later it might present a version mismatch as the link always points to the latest.

**Comment by gsmet** [comment]
I'm unmarking all the doc PRs for backports. They are all part of massive changes that shouldn't be backported. Please don't mark for backport massive structural changes to the doc.

**Comment by gsmet** [comment]
> We need to find another way to get the recent 2.16 relevant enhancements published to

Could you explain why you would want to do that?
Because I don't think it's time well invested: we don't do large changes to the doc of a given minor once it is published.
We never did that and I don't see why it would be necessary to have these changes in 2.16. Unless I'm missing something, in that case, please correct me as it will probably happen a lot in the future.

Chances are that if the PR is very large (which is extremely probable given the PRs I have seen), I won't accept it for a micro anyway.

**Comment by michelle-purcell** [comment]
@gsmet - Because when Quarkus customers click "read the guides" or any doc portal link, they see an older doc state. 
image

They don't see content improvements that folks have done since mid-January, which is a pity.

Most of those content improvements apply also to 2.16 and are being used to source the downstream version of the equivalent guide, which is currently in the 2.x doc project.

**Pull request merged**
